diagnosis/clusterhealthcheck.sh

```diff
@@ -0,0 +1,230 @@
+do
+    HOST_NAME=$(ssh -q -t $USER_NAME@$IP 'echo "$(hostname)"')
+    log_level -i "$HOST_NAME"
+    
+    if [[ $HOST_NAME == *"master"* ]]; then
+        log_level -i "Checking for kubectl on [$HOST_NAME]"
+        KUBECTL_COMMAND=$(ssh -q -t $USER_NAME@$IP "kubectl get nodes")
+        if [[ $KUBECTL_COMMAND == *"NAME"* ]]; then
+            log_level -i "Kubectl is installed and configued on master node. Kubectl test passed"
+        else
+            log_level -e "Kubectl is not installed and configued on master node. Kubectl test failed"
+        fi
+    else
+        log_level -i "Host [$HOST_NAME] is not master, skipping test"
+    fi
+done
+
+# ##############################################################
+# Check if kube-system containers are running in docker on nodes
+
+log_level -i "Checking for kubernetes containers running on nodes"
+MASTER_CONTAINERS="kube-scheduler kube-controller kube-apiserver kube-addon-manager"
+NODE_CONTAINERS="azure-ip-masq-agent kube-flannel kube-proxy"
+for IP in $HOST_LIST
+do
+    HOST_NAME=$(ssh -q -t $USER_NAME@$IP 'echo "$(hostname)"')
+    CONTAINER_LIST=$(ssh -q -t $USER_NAME@$IP 'LIST=$(docker container ls --format="{{.Names }}" --filter status="running"); echo $LIST')
+    
+    if [[ $HOST_NAME == *"master"* ]]; then
+        log_level -i "Checking for master containers"
+        for CONTAINER in $MASTER_CONTAINERS
+        do
+            if [[ $CONTAINER_LIST == *"$CONTAINER"* ]]; then
+                log_level -i "$CONTAINER running on [$HOST_NAME]. Test passed"
+            else
+                log_level -e "$CONTAINER not running on [$HOST_NAME]. Test failed"
+            fi
+        done
+        log_level -i "Checking for node containers"
+        for CONTAINER in $NODE_CONTAINERS
+        do
+            if [[ $CONTAINER_LIST == *"$CONTAINER"* ]]; then
+                log_level -i "$CONTAINER running on [$HOST_NAME]. Test passed"
+            else
+                log_level -e "$CONTAINER not running on [$HOST_NAME]. Test failed"
+            fi
+        done
+    else
+        log_level -i "Checking for node containers"
+        for CONTAINER in $NODE_CONTAINERS
+        do
+            if [[ $CONTAINER_LIST == *"$CONTAINER"* ]]; then
+                log_level -i "$CONTAINER running on [$HOST_NAME]. Test passed"
+            else
+                log_level -e "$CONTAINER not running on [$HOST_NAME]. Test failed"
+            fi
+        done
+    fi
+done
+
+# ##############################################################
+# Check if node to node communication works
+
+log_level -i "Checking nodes can communicate with other nodes"
+for IP in $HOST_LIST
+do
+    HOST_NAME=$(ssh -q -t $USER_NAME@$IP 'echo "$(hostname)"')
+    log_level -i "Checking connectivity from [$HOST_NAME] to cluster nodes"
+    for CURRENTIP in $HOST_LIST
+    do
+        if [[ $IP != $CURRENTIP ]]; then
+            PING_RESULT=$(ssh -q -t $USER_NAME@$IP "ping -c 1 -W 1 "$CURRENTIP" | grep '64 bytes'")
```

Comment:
ICMP is blocked outside the NSG, would that be a problem?

Reply:
Not sure what this means, could you please explain